backend/lib/database.ts

```diff
@@ -46,14 +55,36 @@ export class DatabaseQueries {
       await DatabaseQueries.db
         .update(pixels)
         .set({ color })
-        .where(eq(pixels.id, `${x},${y}`));
+        .where(
+          and(
+            eq(pixels.canvas, DatabaseQueries.activeCanvas),
+            eq(pixels.x, x),
+            eq(pixels.y, y)
+          )
+        );
     }
   }
+
   async setup() {
     const queries = [
-      `CREATE TABLE IF NOT EXISTS ${DATABASE_NAME} (id TEXT PRIMARY KEY, x INTEGER, y INTEGER, color TEXT)`,
+      "CREATE TABLE IF NOT EXISTS `pixels` (`canvas` INTEGER, `x` INTEGER, `y` INTEGER, `color` INTEGER, PRIMARY KEY(`canvas`, `x`, `y`))",
+      "CREATE TABLE IF NOT EXISTS `canvases` (`id` INTEGER NOT NULL, `name` TEXT, PRIMARY KEY(`id`))",
     ] as any;
 
-    await DatabaseQueries.db.run(sql(queries as TemplateStringsArray));
+    DatabaseQueries.db.run(sql(queries as TemplateStringsArray));
+
+    if (!INITIAL_CANVAS) {
+      throw new Error("INITIAL_CANVAS is a required env variable.");
+    } else {
+      const canvas = await DatabaseQueries.db
+        .select()
+        .from(canvases)
+        .where(eq(canvases.name, INITIAL_CANVAS));
+
+      if (canvas.length < 1)
+        throw new Error("No canvas corresponds to INITIAL_CANVAS.");
+      else DatabaseQueries.activeCanvas = canvas[0]!.id;
```

Comment:
Why does the non-null assertion need to be used here?

Reply:
Apparently it doesn't, though imo it kind of should because TS has no knowledge of the length of the array

Reply:
Anyways, removed

Reply:
It's really stupid, which is why in general I do some sort of:
```js
const firstCanvas = canvas.at(0);
if (!firstCanvas) throw....
DatabaseQueries.activeCanvas = firstCanvas;
```
Throw tells typescript to narrow the firstCanvas type from like canvas | undefined to canvas which is what we want; IMO it would be better if the if canvas length check also did narrowing but yeah